code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/PresentableMessageMapper.java

```diff
@@ -0,0 +1,325 @@
+            }
+
+
+            final Map<String, Object> details = consequence.getDetail();
+            if (MapUtils.isNullOrEmpty(details)) {
+                Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "Invalid consequence (%s). Required field \"detail\" is null or empty.", consequence.toString());
+                throw new MessageRequiredFieldMissingException("Required field: \"detail\" is null or empty.");
+            }
+
+            final String schemaType = DataReader.optString(details, MESSAGE_CONSEQUENCE_DETAIL_KEY_SCHEMA, "");
+            if (!SCHEMA_IAM.equals(schemaType)) {
+                Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "Invalid consequence (%s). Required field \"schema\" is (%s) should be of type (%S).", consequence.toString(), schemaType, SCHEMA_IAM);
+                throw new MessageRequiredFieldMissingException("Required field: \"schema\" is not equal to \"https://ns.adobe.com/personalization/message/in-app\".");
+            }
+
+            final Map<String, Object> data = DataReader.optTypedMap(Object.class, details, MESSAGE_CONSEQUENCE_DETAIL_KEY_DATA, null);
+            if (MapUtils.isNullOrEmpty(data)) {
+                Log.debug(MessagingConstants.LOG_TAG, SELF_TAG, "Invalid consequence (%s). Required field \"data\" is null or empty.", consequence.toString());
+                throw new MessageRequiredFieldMissingException("Required field: \"data\" is null or empty.");
+            }
+
+            final String html = DataReader.optString(data, MESSAGE_CONSEQUENCE_DETAIL_KEY_CONTENT, "");
+            if (StringUtils.isNullOrEmpty(html)) {
+                Log.warning(MessagingConstants.LOG_TAG, SELF_TAG, "Unable to create an in-app message, the html payload is null or empty.");
+                throw new MessageRequiredFieldMissingException("Required field: \"html\" is null or empty.");
+            }
+
+            final InAppMessageSettings settings = InAppMessageSettingsFromMap(rawInAppMessageSettings, html, assetMap);
+
+            final UIService uiService = ServiceProvider.getInstance().getUIService();
+            if (uiService == null) {
+                Log.warning(MessagingConstants.LOG_TAG, SELF_TAG, "The UIService is unavailable. Aborting in-app message creation.");
+                throw new IllegalStateException("The UIService is unavailable");
+            }
+            aepMessage = uiService.create(new InAppMessage(settings, new MessagingFullscreenEventListener()), new DefaultPresentationUtilityProvider());
+        }
+
+        /**
+         * Dispatch tracking information via a Messaging request content event.
+         *
+         * @param interaction {@code String} containing the interaction which occurred
+         * @param eventType   {@link MessagingEdgeEventType} enum containing the Event Type to be used for the ensuing Edge Event
+         */
+        public void track(final String interaction, final MessagingEdgeEventType eventType) {
```

Comment:
Looks like this is an overridden method. Can you add the `@Override` annotation for clarity. Same for below methods overridden from `Message`.